src/scripts/stored_procedures.sql

```diff
@@ -501,12 +502,19 @@ gpc_label:begin
 		leave gpc_label;
 	end if;
     
-    insert into posts(user_id, content, img_src, cloudinary_id, post_type) 
-    values (user_id, content, img_src, cloudinary_id, post_type);
+    if post_type = 'group' then
+		insert into posts(user_id, content, img_src, cloudinary_id, post_type) 
+		values (user_id, content, img_src, cloudinary_id, post_type);
+    
+		insert into group_posts(post_id, group_id)
+		values (last_insert_id(), group_id);
+    end if;
+    
+    if post_type = 'shared' then
+		insert into posts(user_id, content, referenced_post_id, post_type) 
+		values (user_id, content, referenced_post_id, post_type);
+    end if;
```

Comment:
Lines from 506 to 510 are correct, keep only those. 

Always insert `post_type` as "group" and create the entry in the `group_posts` table.